File: python/Cpu.py
from Komponen import Komponen
class Cpu(Komponen) :
    def __init__(self, jumlahCore, kecepatanGHz, merk, nama):
        super().__init__(merk, nama)
        
        self.__jumlahCore = jumlahCore
        self.__kecepatanGHz = kecepatanGHz
        
    def get_jumlahCore(self):
        return self.__jumlahCore
    def get_kecepatanGHz(self):
        return self.__kecepatanGHz
    
    # Setter methods
    def set_jumlahCore(self, jumlahCore):
        self.__jumlahCore = jumlahCore
    def set_kecepatanGHz(self, kecepatanGHz):
        self.__kecepatanGHz = kecepatanGHz
        

File: cpp/Cpu.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;
class Cpu : public Komponen {
private:
    int jumlahCore;
    float kecepatanGHz;

public:
    Cpu() {
    }
    Cpu(int jumlahCore, float kecepatanGHz, string merk, string nama) : Komponen(merk, nama) {
        this->jumlahCore = jumlahCore;
        this->kecepatanGHz = kecepatanGHz;
    }
    void setJumlahCore(int jumlahCore) {
        this->jumlahCore = jumlahCore;
    }
    void setKecepatanGHz(float kecepatanGHz) {
        this->kecepatanGHz = kecepatanGHz;
    }
    int getJumlahCore() {
        return this->jumlahCore;
    }

    float getKecepatanGHz() {
        return this->kecepatanGHz;
    }
    ~Cpu() {
    }
};